src/modules/vtol_att_control/tailsitter.cpp

```diff
@@ -152,38 +140,41 @@ Tailsitter::scale_mc_output()
-}
-
-/**
-* Prepare message to acutators with data from mc attitude controller.
-*/
-void Tailsitter::fill_mc_att_control_output()
-{
-	_actuators_out_0->control[0] = _actuators_mc_in->control[0];
-	_actuators_out_0->control[1] = _actuators_mc_in->control[1];
-	_actuators_out_0->control[2] = _actuators_mc_in->control[2];
-	_actuators_out_0->control[3] = _actuators_mc_in->control[3];
-
-	if (_params->elevons_mc_lock == 1) {
-		_actuators_out_1->control[0] = 0;
-		_actuators_out_1->control[1] = 0;
-	} else {
-		_actuators_out_1->control[0] = _actuators_mc_in->control[2];	//roll elevon
-		_actuators_out_1->control[1] = _actuators_mc_in->control[1];	//pitch elevon
+	switch(_vtol_mode) {
+		case ROTARY_WING:
+			_actuators_out_0->control[actuator_controls_s::INDEX_ROLL] = _actuators_mc_in->control[actuator_controls_s::INDEX_ROLL];
+			_actuators_out_0->control[actuator_controls_s::INDEX_PITCH] = _actuators_mc_in->control[actuator_controls_s::INDEX_PITCH];
+			_actuators_out_0->control[actuator_controls_s::INDEX_YAW] = _actuators_mc_in->control[actuator_controls_s::INDEX_YAW];
+			_actuators_out_0->control[actuator_controls_s::INDEX_THROTTLE] = _actuators_mc_in->control[actuator_controls_s::INDEX_THROTTLE];
+
+			if (_params->elevons_mc_lock == 1) {
+				_actuators_out_1->control[0] = 0;
+				_actuators_out_1->control[1] = 0;
+			} else {
+				// NOTE: There is no mistake in the line below, multicopter yaw axis is controlled by elevon roll actuation!
+				_actuators_out_1->control[actuator_controls_s::INDEX_ROLL] = _actuators_mc_in->control[actuator_controls_s::INDEX_YAW];	//roll elevon
+				_actuators_out_1->control[actuator_controls_s::INDEX_PITCH] = _actuators_mc_in->control[actuator_controls_s::INDEX_PITCH];	//pitch elevon
+			}
+			break;
+		case FIXED_WING:
+			// in fixed wing mode we use engines only for providing thrust, no moments are generated
+			_actuators_out_0->control[actuator_controls_s::INDEX_ROLL] = 0;
+			_actuators_out_0->control[actuator_controls_s::INDEX_PITCH] = 0;
+			_actuators_out_0->control[actuator_controls_s::INDEX_YAW] = 0;
+			_actuators_out_0->control[actuator_controls_s::INDEX_THROTTLE] = _actuators_fw_in->control[actuator_controls_s::INDEX_THROTTLE];
+
+			_actuators_out_1->control[actuator_controls_s::INDEX_ROLL] = -_actuators_fw_in->control[actuator_controls_s::INDEX_ROLL];	// roll elevon
+			_actuators_out_1->control[actuator_controls_s::INDEX_PITCH] = _actuators_fw_in->control[actuator_controls_s::INDEX_PITCH] + _params->fw_pitch_trim;	// pitch elevon
+			_actuators_out_1->control[actuator_controls_s::INDEX_YAW] = _actuators_fw_in->control[actuator_controls_s::INDEX_YAW];	// yaw
+			_actuators_out_1->control[actuator_controls_s::INDEX_THROTTLE] = _actuators_fw_in->control[actuator_controls_s::INDEX_THROTTLE];	// throttle
+			break;
+		case TRANSITION:
+		case EXTERNAL:
+			// not yet implemented, we are switching brute force at the moment
+			break;
```

Comment:
Doesn't this complain that there is no default? If not than, I am complaining :).